context/app/static/js/components/Search/SearchWrapper.jsx

```diff
@@ -46,31 +44,61 @@ function getByPath(nested, field) {
   return current;
 }
 
-function makeTableComponent(resultFields, detailsUrlPrefix, idField) {
-  return function ResultsTable(props) {
+function makeTheadComponent() {
+  return function ResultsThead(props) {
+    const { items, toggleItem, selectedItems } = props;
+    if (selectedItems.length > 1) {
+      console.warn('Expected only a single sort, not:', selectedItems);
+    }
+    const selectedItem = selectedItems.length ? selectedItems[0] : undefined;
+    const pairs = [];
+    for (let i = 0; i < items.length; i += 2) {
+      const pair = items.slice(i, i + 2);
+      pairs.push(pair);
+      if (pair[0].label !== pair[1].label || pair[0].field !== pair[1].field) {
+        console.warn('Expected pair.label and .field to match', pair);
+      }
+    }
+    return (
+      <thead>
+        <tr>
+          {pairs.map((pair) => {
+            const match = pair.filter((item) => item.key === selectedItem);
+            const order = match.length ? match[0].order : undefined;
+            const orderIcon = order ? { asc: '▲', desc: '▼' }[order] : undefined;
+            return (
+              <th
+                key={pair[0].key}
+                onClick={() => {
+                  toggleItem(pair[order && order === pair[0].order ? 1 : 0].key);
+                }}
+              >
+                {pair[0].label}&nbsp;{orderIcon}
+              </th>
```

Comment:
Good thought -- I've just add `role="button"`... but if that's not really enough, let me know? Want to avoid more nesting elements / more css if possible.